crates/sui-indexer-alt-consistent-store/src/rpc/middleware/metrics.rs

```
@@ -0,0 +1,151 @@
+            .start_timer();
+
+        metrics
+            .requests_received
+            .with_label_values(&[path.as_ref()])
+            .inc();
+
+        Handler {
+            metrics,
+            path,
+            timer: Some(timer),
+        }
+    }
+}
+
+impl ResponseHandler for Handler {
+    fn on_response(&mut self, response: &http::response::Parts) {
+        const GRPC_STATUS: http::HeaderName = http::HeaderName::from_static("grpc-status");
+
+        let status = if response
+            .headers
+            .get(&CONTENT_TYPE)
+            .is_some_and(|content_type| {
+                content_type
+                    .as_bytes()
+                    // check if the content-type starts_with 'application/grpc' in order to
+                    // consider this as a gRPC request. A prefix comparison is done instead of a
+                    // full equality check in order to account for the various types of
+                    // content-types that are considered as gRPC traffic.
+                    .starts_with(GRPC_CONTENT_TYPE.as_bytes())
+            }) {
+            let code = response
+                .headers
+                .get(&GRPC_STATUS)
+                .map(http::HeaderValue::as_bytes)
+                .map(Code::from_bytes)
+                .unwrap_or(Code::Ok);
+
+            code_as_str(code)
+        } else {
+            response.status.as_str()
+        };
+
+        self.timer.take().map(HistogramTimer::stop_and_record);
+
+        if status == "ok" {
+            self.metrics
+                .requests_succeeded
+                .with_label_values(&[self.path.as_ref(), status])
+                .inc();
+        } else {
+            self.metrics
+                .requests_failed
+                .with_label_values(&[self.path.as_ref(), status])
+                .inc();
+        }
+    }
+
+    fn on_error<E>(&mut self, _error: &E) {
+        // Do nothing if the whole service errored
```

Can you add this comment in a panic in case it somehow unexpectedly triggers?